Orange/tests/test_distribution.py

```diff
@@ -338,93 +338,82 @@ def test_get_distributions(self):
         np.testing.assert_almost_equal(ddist[-1], [50, 50, 50])
 
     def test_sparse_get_distributions(self):
-        def assert_dist_and_unknowns(computed, gold_dist):
+        def assert_dist_and_unknowns(computed, goal_dist):
             nonlocal d
-            gold_dist = np.array(gold_dist)
-            sum_dist = np.sum(gold_dist[1, :] if gold_dist.ndim == 2 else gold_dist)
+            goal_dist = np.array(goal_dist)
+            sum_dist = np.sum(goal_dist[1, :] if goal_dist.ndim == 2 else goal_dist)
             n_all = np.sum(d.W) if d.has_weights() else len(d)
 
-            np.testing.assert_almost_equal(computed, gold_dist)
+            np.testing.assert_almost_equal(computed, goal_dist)
             self.assertEqual(computed.unknowns, n_all - sum_dist)
 
         domain = data.Domain(
-            [data.DiscreteVariable("d%i" % i, values=list("abc"))
-             for i in range(10)] +
+            [data.DiscreteVariable("d%i" % i, values=list("abc")) for i in range(10)] +
             [data.ContinuousVariable("c%i" % i) for i in range(10)])
 
-        #  0  1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19
-        # ------------------------------------------------------------
-        #     2     2  1  1  2        1           1  1     2  0  2
-        #        1  1  0  0  1     2                 2     1  0
-        #           1     2  0
-        #
-        #        2        0  1                   1.1
-        #
-        sdata = np.array([2, 2, 1, 1, 2, 1, 1, 1, 2, 0, 2,
-                          1, 1, 0, 0, 1, 2, 2, 1, 0,
-                          1, 2, 0,
-                          2, 0, 1, 1.1])
-        indices = [1, 3, 4, 5, 6, 9, 13, 14, 16, 17, 18,
-                   2, 3, 4, 5, 6, 8, 14, 16, 17,
-                   3, 5, 6,
-                   2, 5, 6, 13]
-        indptr = [0, 11, 20, 23, 23, 27]
-        X = sp.csr_matrix((sdata, indices, indptr), shape=(5, 20))
+        X = sp.csr_matrix(
+            # 0  1  2  3       4       5       6  7  8  9 10 11 12   13 14 15 16      17 18 19
+            # --------------------------------------------------------------------------------
+            [[0, 2, 0, 2,      1,      1,      2, 0, 0, 1, 0, 0, 0,   1, 1, 0, 2, np.nan, 2, 0],
+             [0, 0, 1, 1, np.nan, np.nan,      1, 0, 2, 0, 0, 0, 0,   0, 2, 0, 1, np.nan, 0, 0],
+             [0, 0, 0, 1,      0,      2, np.nan, 0, 0, 0, 0, 0, 0,   0, 0, 0, 0,      0, 0, 0],
+             [0, 0, 0, 0,      0,      0,      0, 0, 0, 0, 0, 0, 0,   0, 0, 0, 0,      0, 0, 0],
+             [0, 0, 2, 0,      0,      0,      1, 0, 0, 0, 0, 0, 0, 1.1, 0, 0, 0,      0, 0, 0]]
+        )
```

Comment:
This test doesn't quite work for explicit zeros. E.g. if you add `X[0,0] = 0` here, the test should pass and the results should be the same — but it fails. IMO the problem is in the `bincount`, check the comment above.

Reply:
Also, once this is corrected, please add some explicit zeros to this example.